### PR TITLE
CY8CKIT-064S0S2-4343W: BLE integration

### PR DESCRIPTION
**Pull request by RaymondNgun**
Adds BLE support to Cypress CY8CKIT-064S0S2-4343W kit

Description
-----------
This change adds BLE support to CY8CKIT-064S0S2-4343W target. Note that BLE is
enabled via -DBLE_SUPPORTED=1 in the compilation command.

Checklist:
----------
- [x] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vendors/cypress/bluetooth/include/wiced_bt_nvram_access.h [hunk 1]
```diff
@@ -170,4 +170,4 @@ wiced_result_t wiced_bt_nvram_access_register(wiced_bt_nvram_access_t *p_nvram_a
 
 #ifdef __cplusplus
 }
-#endif
+#endif
```
Comment (gkwicker) [inline vendors/cypress/bluetooth/include/wiced_bt_nvram_access.h:173]: It looks like there are extra characters at the end of this line, can you remove them?

File: vendors/cypress/bluetooth/psoc6/cyosal/src/cybt_osal_amzn_freertos.c [hunk 2]
```diff
@@ -50,11 +50,9 @@ const uint32_t millis_to_tick_ratio = (uint32_t)( 1000 / configTICK_RATE_HZ );
  *               Function Definitions
  ******************************************************/
 
-extern signed portBASE_TYPE xTaskIsTaskFinished( TaskHandle_t xTask ) PRIVILEGED_FUNCTION;
-
 wiced_result_t wiced_rtos_init_semaphore( wiced_semaphore_t* sem )
 {
-    SemaphoreHandle_t xsemaphore = NULL;
+    SemaphoreHandle_t* xsemaphore = NULL;
```
Comment (gkwicker) [inline vendors/cypress/bluetooth/psoc6/cyosal/src/cybt_osal_amzn_freertos.c:55]: Why was this changed?  `xSemaphoreCreateCounting()` (line 61) [returns `SemaphoreHandle_t`](https://www.freertos.org/CreateCounting.html), not `SemaphoreHandle_t *`.  It looks like this line should be changed back to the original version.

File: vendors/cypress/bluetooth/include/wiced_bt_nvram_access.h (hunk 1 above)
Comment (RaymondNgun) [reply vendors/cypress/bluetooth/include/wiced_bt_nvram_access.h:173]: Thanks. Will fix.

File: vendors/cypress/bluetooth/psoc6/cyosal/src/cybt_osal_amzn_freertos.c (hunk 2 above)
Comment (RaymondNgun) [reply vendors/cypress/bluetooth/psoc6/cyosal/src/cybt_osal_amzn_freertos.c:55]: You are correct. Will Fix. Thanks.

**Comment by RaymondNgun** [comment]
@ravibhagavandas, BLE_SUPPORTED is now enabled by default for this board.
